src/wp-includes/option.php

```diff
@@ -3214,3 +3214,126 @@ function wp_autoload_values_to_autoload() {
 
 	return array_intersect( $filtered_values, $autoload_values );
 }
+
+
+/**
+ * Filters site metadata for multisite installations.
+ *
+ * This function is specific to multisite setups and provides a way to retrieve
+ * network options. It enforces parameters' constraints and suggests using an alternative function.
+ *
+ * @param mixed $current The value to return if no changes are made.
+ * @param int $object_id Site ID.
+ * @param string $meta_key Metadata key.
+ * @param bool $single Whether to retrieve a single value. Must be true.
+ *
+ * @return mixed The network option value for the specified meta key, or false if the function is used incorrectly.
+ */
+function filter_get_site_metadata( $current, $object_id, $meta_key, $single ) {
+	if ( ! is_multisite() ) {
+		_doing_it_wrong( 'get_metadata', __( 'This function is only available in multisite installations.' ), '6.8.0' );
+		return false;
+	}
```

Comment:
On non-multisite, `get_network_option()` still works and falls through to `get_option()`. So it would make sense to me to align with that. We could remove this condition since `get_network_option()` already handles it.

Reply:
I think should do that. currently, `get_metadata( 'site', 1, 'site_url' )` in a single site, would result false, as it should. It is start returning the option, than that could be considered a breaking change.

Reply:
I don't think it's a breaking change to return another value - the function is expected to return _anything_, so returning anything is not a breaking change. If you argue that way, even using `get_network_option()` here at all would be a breaking change.

Reply:
> I don't think it's a breaking change to return another value - the function is expected to return _anything_, so returning anything is not a breaking change. If you argue that way, even using `get_network_option()` here at all would be a breaking change.

I am confused by this feedback. Currently if I called `get_metadata( 'site', 1, 'site_url' )`, what happens if is try to load data from the sitemeta table. If table doesn't exist, it will result false. If we use `get_network_option` it will load from option the response of the function call goes from false to option value. That is a different and breaking change for the code.